### PR TITLE
Bce with logits

### PR DESCRIPTION
**Pull request by sevmag**
PR addressing issue #846

### CODE REVIEW

**Review by Aske-Rosted — COMMENTED** [review]
Looks good although it seems like a some unrelated formatting changes were added to the LogCMK class, was this intended?

**Comment by sevmag** [comment]
Uf, good catch, thanks! Did not pay too much attention when resolving the merge conflict. Lesson learned. Should be good now.

**Review by Aske-Rosted — APPROVED** [review]
looks good to me 👍
